internal/dataplane/sendconfig/inmemory_schema_test.go

```diff
@@ -0,0 +1,299 @@
+package sendconfig_test
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/kong/deck/file"
+	"github.com/kong/go-kong/kong"
+	"github.com/stretchr/testify/require"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane/sendconfig"
+)
+
+func TestDBLessConfigMarshalToJSON(t *testing.T) {
+	dblessConfig := sendconfig.DBLessConfig{
+		Content: file.Content{
+			Services: []file.FService{
+				{
+					Service: kong.Service{
+						Name: kong.String("service-id"),
+					},
+				},
+			},
+		},
+		ConsumerGroupConsumerRelationships: []sendconfig.ConsumerGroupConsumerRelationship{
+			{
+				ConsumerGroup: "cg1",
+				Consumer:      "c1",
+			},
+		},
+		ConsumerGroupPluginRelationships: []sendconfig.ConsumerGroupPluginRelationship{
+			{
+				ConsumerGroup: "cg1",
+				Plugin:        "p1",
+			},
+		},
+	}
+
+	expected := `{
+  "services": [
+    {
+      "name": "service-id"
+    }
+  ],
+  "consumer_group_consumers": [
+    {
+      "consumer_group": "cg1",
+      "consumer": "c1"
+    }
+  ],
+  "consumer_group_plugins": [
+    {
+      "consumer_group": "cg1",
+      "plugin": "p1"
+    }
+  ]
+}`
+	b, err := json.MarshalIndent(dblessConfig, "", "  ")
+	require.NoError(t, err)
+	require.Equal(t, expected, string(b))
```

Comment:
Maybe to not rely on indentation, etc. let's use [require.JSONEq](https://pkg.go.dev/github.com/stretchr/testify/require#JSONEq)